src/nodejs/src/logging.ts

```diff
@@ -4,10 +4,67 @@ import 'logging-init';
 export { LogLevel } from './logging-init';
 export type { LogScope } from './logging-init';
 
+export interface LogRef {
+    target : unknown;
+    id : number;
+}
+
+interface SetItem {
+    ref : LogRef;
+    touchedAt : number;
+}
+
+class LogRefSet {
+    items : SetItem[];
+    capacity : number;
+    idSeed : number;
+
+    constructor(capacity : number) {
+        this.idSeed = 0;
+        this.capacity = capacity;
+        this.items = [];
+    }
+
+    public ref(data: unknown) : LogRef {
+        const itemIndex = this.items.findIndex(el => el.ref.target === data);
+        if (itemIndex >= 0) {
+            const existentItem = this.items[itemIndex];
+            existentItem.touchedAt = Date.now();
+            return existentItem.ref;
+        }
+        else {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+            const id = data['__logRefId'] as number ?? this.idSeed++;
```

Comment:
why do we mutate input parameter?

Reply:
what happens if string, tuple or smth like that is passed as data?

Reply:
changed type of input argument to object